test_settings.py

```diff
@@ -28,6 +29,16 @@
     }
 }
 
+if 'DATABASE_URL' in os.environ:
+    try:
+        import dj_database_url
+        import psycopg2
+        DATABASES['default'] = dj_database_url.config()
+    except ImportError:
+        print('Using the DATABASE_URL variable requires dj-database-url and '
+              'psycopg2. Try:\n\npip install -r travis.txt')
+        sys.exit(1)
```

Comment:
My feeling is that it's better to error and exit in this scenario rather than falling back to a database backend that isn't what was specified for the environment.

Reply:
You could also either (a) let the exception bubble up or (b) re-raise after printing the error message. Either way, an exception at this point will halt execution.

Reply:
Agreed, better not to call `sys.exit()` and let it exit on its own by re-raising.

I don't love the implicit dependency on `psycopg2` just because someone has defined `DATABASE_URL` at all—it might be useful to run the tests against MariaDB or to use PyGreSQL. But I don't think it's worth blocking this on.

Reply:
I agree re: the dependency on psycopg2. I was taking a "Don't build it until you need it" approach, figuring that the next person who comes along and wants to run tests locally with a different database could submit their own pull request to make the logic to handle the `ImportError` more surgical. Perhaps that was selfish of me.

I'm happy to make the logic here a little more conservative, only requiring psycopg2 if the DB backend is actually Postgres.